src/core/bluetoothdevicemodel.cpp

```diff
@@ -16,44 +16,60 @@
 
 #include "bluetoothdevicemodel.h"
 #include "qgis.h"
+#include <QSettings>
 #include <QDebug>
 
 BluetoothDeviceModel::BluetoothDeviceModel( QObject *parent )
   : QAbstractListModel( parent ),
     mLocalDevice( std::make_unique<QBluetoothLocalDevice>() )
 {
   connect( &mServiceDiscoveryAgent, &QBluetoothServiceDiscoveryAgent::serviceDiscovered, this, &BluetoothDeviceModel::serviceDiscovered );
-  connect( &mServiceDiscoveryAgent, qgis::overload<QBluetoothServiceDiscoveryAgent::Error>::of( &QBluetoothServiceDiscoveryAgent::error ), [ = ]()
+  connect( &mServiceDiscoveryAgent, qgis::overload<QBluetoothServiceDiscoveryAgent::Error>::of( &QBluetoothServiceDiscoveryAgent::error ), this, [ = ]()
   {
     setLastError( mServiceDiscoveryAgent.errorString() );
     setScanningStatus( Failed );
     endResetModel();
   } );
   connect( &mServiceDiscoveryAgent, &QBluetoothServiceDiscoveryAgent::finished, [ = ]()
   {
-    setScanningStatus( Succeeded );
+    setScanningStatus( mServiceDiscoveryAgent.error() == QBluetoothServiceDiscoveryAgent::NoError ? Succeeded : Failed );
     endResetModel();
   } );
   connect( &mServiceDiscoveryAgent, &QBluetoothServiceDiscoveryAgent::canceled, [ = ]()
   {
     setScanningStatus( Canceled );
     endResetModel();
   } );
+
+  beginResetModel();
+  mDiscoveredDevices.clear();
+  mDiscoveredDevices.append( qMakePair( tr( "Internal device" ), QString() ) );
+
+  QSettings settings;
+  const QString deviceAddress = settings.value( QStringLiteral( "positioningDevice" ), QString( "" ) ).toString();
```

Comment:
```suggestion
  const QString deviceAddress = settings.value( QStringLiteral( "positioningDevice" ) ).toString();
```

For an empty string use `QString()`. If you skip it here, that's done automatically.